### PR TITLE
feat(messaging): Support Android 13+ notification permissions

### LINKED ISSUE #321: [Bug][Messaging] - W/FirebaseMessaging: Missing Default Notification Channel metadata in AndroidManifest. Default value will be used.
**Plugin(s):**
  "@capacitor/cli": "^4.6.1",
  "@capacitor/core": "^4.6.1",
  "@capacitor-firebase/messaging": "^1.3.0",

**Platform(s):**
Android

**Current behavior:**
I try to request permission, get an answer that "permission is granted", but than wont see the app in the Notification list in the settings. Of course no message would appear. 
after sending a notification from firebase console - i get the above error in android studio:
```W/FirebaseMessaging: Missing Default Notification Channel metadata in AndroidManifest. Default value will be used.```

when i reopen the app - the prompt is shown - yet the notification wont appear -from here and on everything works correctly.

image

image

**Expected behavior:**
Prompt should appear on start, as declared in the code.

**Steps to reproduce:**
add this code to index.js in a react app:

```
requestPermissions();
subscribeToTopic("testing");
```

and open the app.
now send a message from fb console to topic "testing" and wait for the log to appear in android studio
now go to the app...

**Related code:**
mentioned above :P

**Other information:**
Android 13
Pixel 6a

**Capacitor doctor:**

```
💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: 4.6.3
  @capacitor/core: 4.6.3
  @capacitor/android: 4.6.3
  @capacitor/ios: 4.6.3

Installed Dependencies:

  @capacitor/cli: 4.6.1
  @capacitor/core: 4.6.1
  @capacitor/ios: 4.6.1
  @capacitor/android: 4.6.1

[success] iOS looking great! 👌
[error] Missing <manifest package=""> attribute in app/src/main
```

### PR DESCRIPTION
**Pull request by CalumMurray**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] The changes have been tested successfully.
- [x] A changeset has been created (`npm run changeset`).

Close #321

### CODE REVIEW

**Comment by robingenz** [comment]
The permission `POST_NOTIFICATIONS` is only supported from Android 13 (API Level 33).
Capacitor 4 currently officially supports API level 32 (see [this comment](https://github.com/ionic-team/capacitor-plugins/issues/1116#issuecomment-1217720273)). API level 33 will be supported with Capacitor 5. For this reason, this PR has to wait until the release of Capacitor 5. In the meantime you can try to use the following plugin: https://github.com/hermitdemschoenenleben/capacitor-plugin-android-post-notifications-permission

**Comment by CalumMurray** [comment]
Thanks very much for the quick response. I've updated to hopefully fix the build error for when it comes time to merge.

As noted in some other comments, the problem is that targeting API level 32 and using an Android 13 device means that the permission is immediately prompted on first open, meaning we can't prime the user for permissions as is best practice. Do you have any insight into why using a forked version with this permission might cause any issues with Capacitor 4 running on older and newer Androids? Is there some feature in Android 13 that is incompatible with Capacitor 4?

**Comment by robingenz** [comment]
Unfortunately not. I would just give it a try and see if it works. I just can't merge this before Capacitor 5 is released because everyone would need to target API Level 33.

**Comment by CalumMurray** [comment]
Done. It's been a while since I looked at this, so let me know if it still looks alright.

File: .changeset/three-llamas-appear.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@capacitor-firebase/messaging": minor
+---
+
+feat(messaging): Support Android 13+ notification permissions
```
Comment (robingenz) [inline .changeset/three-llamas-appear.md:5]:
```suggestion
feat(android): support Android 13+ notification permission
```

Comment (robingenz) [inline .changeset/three-llamas-appear.md:5]:
```suggestion
feat(android): support Android 13+ notification permission
```

**Review by robingenz — CHANGES_REQUESTED** [review]
The last thing that is missing: we need to check the Android version and should only request the permission if it is >= Android 13.

Example: https://github.com/ionic-team/capacitor-plugins/blob/main/push-notifications/android/src/main/java/com/capacitorjs/plugins/pushnotifications/PushNotificationsPlugin.java#L78:L100

**Comment by robingenz** [comment]
@CalumMurray I just implemented it in another PR so I can finally test the Capacitor 5 version. Thanks for the PR anyway!

**Pull request closed without merging**